### DISCUSSION TITLE
Problem with AI Thinker ESP32 Audio Kit audio routing?

### DISCUSSION THREAD

**Discussion by hlavamir** [pschatzmann/arduino-audio-tools#897] [Q&A] [answered]
Hi Phil and the community here!

Firstly, I would like to thank you for all the amazing work you have done, while creating this library! It's really well thought through and easy to use. Wow!

To my question - I have just purchased an AI Thinker ESP32 Audio Kit board (v2.2 A247, ES8388), but no matter what I tried so far, I haven't been able to get the right channel of the headphones output to work. Always only left channel plays. At the same time the Line In only seems to give me the right channel, left is silent. Mic1 works, Mic2 is silent.

The examples I have tried to build were for example "streams-synthbasic1-audiokit", or "streams-audiokit-audiokit".

I am pretty sure it's not a headphone/cable problem, I have tried multiple. My board should be the #5 AudioKit, at least it's the only one that does works. I have found the workarounds in the AudioKitSettings.h, but did not manage to find a working solution.

So..... at the moment I assume the board I have behaves differently than what one would logically expect, but that it's software-fixable? Can I kindly ask for recommendations, when it comes to the best settings to have stereo line input going directly into the stereo headphone output, microphones muted?

Thanks in advance for any input.... :)

Have a nice day!

**Comment by pschatzmann** [comment, marked as answer]
The simplest test is using the [sine output test](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/streams-generator-audiokit/streams-generator-audiokit.ino)
If this also produces only output on one channel then I tend to think this is rather an issue with the hardware of your board.
- Did you try to switch the speakers: Maybe one is broken.
- Did you try the output via the headphones

**Reply by hlavamir** [reply]
Hi Phil, 
thank you for such a quick reply! I finally got to trying some more things today. When I deployed the "sine output test" example you recommended, both speakers are playing fine, but I still only hear left channel in the headphones. I am worried it could be a hardware problem as you say, but the board arrived by post well packed and does not show any signs of damage...

I also tried simple audio-through (streams-audiokit-audiokit) and the board seems to do following
- both speakers are playing what goes into the line input, but if I try to "pan" the input, I get interesting results - left channel only I hear correctly on the left speaker, right channel only plays on both speakers
- left microphone is always on, right one off
- headphones left channel is playing, right one not

All that is with default settings for board #5 in the AudioKitSettings.h

I was hoping you might have encountered something like this in the past with your experience and could tell me it's just some config issue... But I'm starting to feel like it's an issue with the board itself.

Maybe a stupid question, but I could not find an answer anywhere - what is the best board to select in Arduino IDE? So far I have been using ESP32 Dev Module, is there any better one?

Thank you!

**Comment by pschatzmann** [comment]
Never seen any problem like this! Maybe the headphones are broken ?
I am also using ESP32 Dev Module

**Reply by usernamepasswordBS** [reply]
Thank you for this answer. I looked for which board to use but could not find it.
